File: 2293. Min Max Game/Solution.py
from typing import List


class Solution:
    def minMaxGame(self, nums: List[int]) -> int:
        n = len(nums)
        while n != 1:
            for i in range(n // 2):
                if i & 1:
                    nums[i] = max(nums[i << 1], nums[i << 1 | 1])
                else:
                    nums[i] = min(nums[i << 1], nums[i << 1 | 1])
            n >>= 1
        return nums[0]

File: 2293. Min Max Game/Solution.cpp
#include <vector>
using namespace std;

class Solution {
public:
    int minMaxGame(vector<int>& nums)
    {
        for (int n = nums.size(); n != 1; n >>= 1) {
            for (int i = 0; i < n / 2; i++) {
                if (i & 1)
                    nums[i] = max(nums[i << 1], nums[i << 1 | 1]);
                else
                    nums[i] = min(nums[i << 1], nums[i << 1 | 1]);
            }
        }
        return nums[0];
    }
};
